tests/unit/moduleapi/async_rm_call.tcl

```diff
@@ -0,0 +1,157 @@
+            set rd [redis_deferring_client]
+
+            $rd $cmd bzpopmin s 0
+            wait_for_blocked_client
+            r zadd s 10 foo
+            assert_equal [$rd read] {s foo 10}
+        }
+
+        test "Bzpopmax on async RM_Call using $cmd" {
+            set rd [redis_deferring_client]
+
+            $rd $cmd bzpopmax s 0
+            wait_for_blocked_client
+            r zadd s 10 foo
+            assert_equal [$rd read] {s foo 10}
+        }
+    }
+
+    test {Nested async RM_Call} {
+        set rd [redis_deferring_client]
+
+        $rd do_rm_call_async do_rm_call_async do_rm_call_async do_rm_call_async blpop l 0
+        wait_for_blocked_client
+        r lpush l a
+        assert_equal [$rd read] {l a}
+    }
+
+    test {Test multiple async RM_Call waiting on the same event} {
+        set rd1 [redis_deferring_client]
+        set rd2 [redis_deferring_client]
+
+        $rd1 do_rm_call_async do_rm_call_async do_rm_call_async do_rm_call_async blpop l 0
+        $rd2 do_rm_call_async do_rm_call_async do_rm_call_async do_rm_call_async blpop l 0
+        wait_for_blocked_client
+        r lpush l element element
+        assert_equal [$rd1 read] {l element}
+        assert_equal [$rd2 read] {l element}
+    }
+
+    test {async RM_Call calls RM_Call} {
+        assert_equal {PONG} [r do_rm_call_async do_rm_call ping]
+    }
+
+    test {async RM_Call calls background RM_Call calls RM_Call} {
+        assert_equal {PONG} [r do_rm_call_async do_bg_rm_call do_rm_call ping]
+    }
+
+    test {async RM_Call calls background RM_Call calls RM_Call calls async RM_Call} {
+        assert_equal {PONG} [r do_rm_call_async do_bg_rm_call do_rm_call do_rm_call_async ping]
+    }
+
+    test {async RM_Call inside async RM_Call callback} {
+        set rd [redis_deferring_client]
+        $rd wait_and_do_rm_call blpop l 0
+        wait_for_blocked_client
+
+        start_server {} {
+            test "Connect a replica to the master instance" {
+                r slaveof [srv -1 host] [srv -1 port]
+                wait_for_condition 50 100 {
+                    [s role] eq {slave} &&
+                    [string match {*master_link_status:up*} [r info replication]]
+                } else {
+                    fail "Can't turn the instance into a replica"
+                }
+            }
+
+            assert_equal {1} [r -1 lpush l a]
+            assert_equal [$rd read] {l a}
+        }
+    }
+
+    test {Become replica while having async RM_Call running} {
+        set rd [redis_deferring_client]
+        $rd wait_and_do_rm_call blpop l 0
+        wait_for_blocked_client
+
+        #become a replica of a not existing redis
+        r replicaof localhost 30000
+
+        catch {[$rd read]} e
+        assert_match {UNBLOCKED force unblock from blocking operation*} $e
+
+        r replicaof no one
+    }
+
+    test {Pipeline with blocking RM_Call} {
+        set rd [redis_deferring_client]
+        $rd do_rm_call_async blpop l 0
```

Comment:
```suggestion
        set buf ""
        append buf "do_rm_call_async blpop l 0\r\n"
        append buf "ping\r\n"
        $rd write $buf
        $rd flush
```

Reply:
Shouldn't we write here a resp protocol?

Reply:
I don't think so. we also do that in other pipeline tests (https://github.com/redis/redis/blob/unstable/tests/unit/type/list.tcl#L2202-L2206)

Reply:
@oranagra I think that is what I offered, no?